client/components/site-preview-link/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Spinner } from '@automattic/components';
+import styled from '@emotion/styled';
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+import ClipboardButtonInput from '../clipboard-button-input';
+import { useCreateSitePreviewLink } from './use-create-site-preview-link';
+import { useDeleteSitePreviewLink } from './use-delete-site-preview-link';
+import { useSitePreviewLinks } from './use-site-preview-links';
+import type { SiteId } from 'calypso/types';
+
+interface SitePreviewLinkProps {
+	siteId: SiteId;
+	siteUrl: string;
+	source: 'launch-settings' | 'privacy-settings' | 'smp-modal';
+	disabled?: boolean;
+	forceOff?: boolean;
+}
+
+const HelpText = styled.p( {
+	display: 'block',
+	margin: '5px 0',
+	fontSize: '0.875rem',
+	fontStyle: 'italic',
+	fontWeight: 400,
+	color: 'var(--color-text-subtle)',
+} );
+
+const NOTICE_OPTIONS = {
+	duration: 5000,
+};
+
+function useDispatchNotice() {
+	const dispatch = useDispatch();
+
+	return {
+		showSuccessNotice: ( message: string ) => dispatch( successNotice( message, NOTICE_OPTIONS ) ),
+		showErrorNotice: ( message: string ) => dispatch( errorNotice( message, NOTICE_OPTIONS ) ),
+	};
+}
+
+export default function SitePreviewLink( {
+	siteId,
+	siteUrl,
+	disabled = false,
+	forceOff = false,
+	source,
+}: SitePreviewLinkProps ) {
+	const translate = useTranslate();
+	const { showSuccessNotice, showErrorNotice } = useDispatchNotice();
+	const [ checked, setChecked ] = useState( false );
+
+	const { data: previewLinks, isLoading: isFirstLoading } = useSitePreviewLinks( {
```

Comment:
Could just default this to an empty array to avoid guarding against non-empty later. `data: previewLinks = []`